src/cascade/input_data/db/configuration.py

```diff
@@ -2,30 +2,111 @@
+    if len([c for c in [meid, mvid, settings_file] if c is not None]) != 1:
+        raise ValueError(
+            "Must supply exactly one of mvid, meid or settings_file")
+    if meid:
+        raw_settings, found_mvid = load_raw_settings_meid(meid)
+    elif mvid:
+        raw_settings, found_mvid = load_raw_settings_mvid(mvid)
+    elif settings_file:
+        raw_settings, found_mvid = load_raw_settings_file(settings_file)
+    else:
+        raise RuntimeError(f"Either meid, mvid, or file must be specified.")
 
+    return json_settings_to_frozen_settings(raw_settings, found_mvid)
 
-def settings_for_model(meid=None, mvid=None):
-    if len([c for c in [meid, mvid] if c is not None]) != 1:
-        raise ValueError("Must supply either mvid or meid")
 
-    if mvid is not None:
-        mvid = mvid
-    else:
-        modelable_entity_id = meid
-        ec = make_execution_context(modelable_entity_id=modelable_entity_id)
-        mvid = latest_model_version(ec)
-        CODELOG.info(f"No model version specified so using the latest version for model {meid} which is {mvid}")
+def load_raw_settings_meid(modelable_entity_id):
+    """
+    Given a meid, get settings for the latest corresponding mvid.
+
+    Args:
+        modelable_entity_id (int,str): The MEID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID that is the latest one associated with this meid.
+    """
+    ec = make_execution_context(modelable_entity_id=modelable_entity_id)
+    mvid = latest_model_version(ec)
+    MATHLOG.info(
+        f"No model version specified so using the latest version for "
+        f"model {modelable_entity_id} which is {mvid}")
+    ec = make_execution_context(model_version_id=mvid)
+    raw_settings = settings_json_from_epiviz(ec)
+    return raw_settings, mvid
+
+
+def load_raw_settings_mvid(mvid):
+    """Given an mvid, get its settings.
+
+    Args:
+        mvid (int,str): Model version ID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID that was passed in.
+    """
     ec = make_execution_context(model_version_id=mvid)
     raw_settings = settings_json_from_epiviz(ec)
+    return raw_settings, mvid
+
+
+def load_raw_settings_file(settings_file):
+    """Given a settings file, get the latest mvid for its meid.
 
-    if "model_version_id" not in raw_settings["model"] or not raw_settings["model"]["model_version_id"]:
+    Args:
+        settings_file (str): Model version ID.
+    Returns:
+        dict: Settings as a JSON dictionary of dictionaries.
+        int: Model version ID, either found in file or latest from meid.
+    """
+    with open(str(settings_file), "r") as f:
+        raw_settings = json.load(f)
+    if "model" in raw_settings and "modelable_entity_id" in raw_settings["model"]:
+        ec = make_execution_context(modelable_entity_id=raw_settings["model"]["modelable_entity_id"])
```

Comment:
Again, this should take an ExecutionContext as an argument and then add the mvid to it.